File: crates/adapter-tree-sitter/tests/fixtures/sample.cpp
class Calculator {
public:
    int compute(int value) {
        auto positive = [](int item) {
            if (item > 0) { return true; }
            return false;
        };
        if (value > 1 && value != 3) {
            return positive(value + 1);
        }
        return 0;
    }
};

File: crates/adapter-tree-sitter/tests/fixtures/sample.rs
struct Thing;

impl Thing {
    fn choose(&self, value: i32) -> bool {
        fn nested(item: i32) -> bool {
            if item > 0 { true } else { false }
        }
        let positive = |item: i32| {
            if item > 0 { true } else { false }
        };
        if value > 1 && value != 3 {
            positive(value + 1)
        } else {
            false
        }
    }
}

/// duplicate marker true == false
